keras_core/losses/losses.py

```diff
@@ -1682,13 +1693,31 @@ def sparse_categorical_crossentropy(y_true, y_pred, from_logits=False, axis=-1):
     >>> loss
     array([0.0513, 2.303], dtype=float32)
     """
-    return ops.sparse_categorical_crossentropy(
+
+    if ignore_class is not None:
+        res_shape = ops.shape(y_pred)[:-1]
+        valid_mask = ops.not_equal(y_true, ops.cast(ignore_class, y_pred.dtype))
+        y_true = y_true * ops.cast(valid_mask, y_true.dtype)
+        y_pred = y_pred * ops.cast(
+            ops.expand_dims(valid_mask, -1), y_pred.dtype
+        )
+
+    res = ops.sparse_categorical_crossentropy(
         y_true,
         y_pred,
         from_logits=from_logits,
         axis=axis,
     )
 
+    if ignore_class is not None:
+        valid_mask = ops.reshape(valid_mask, res_shape)
+        res = ops.where(valid_mask, res, 0.0)
+
+        if backend.backend() != "numpy":
```

Comment:
Rather than using this conditional, do a try/except AttributeError.

Reply:
If we use `try/except`, and use a print statement inside it, the function will have a side-effect when jit compiled. Is there any specific reason for not using `if` here?

Reply:
Try/except is not a side effect and there is no print statement?

Reply:
Okay, so I misunderstood. I thought we need to add a message as well. Thanks for the clarification. I will make the change